src/handlers/messageHandler.ts

```diff
@@ -124,10 +124,15 @@ async function autoCreateThread(message: Message, requestId: Snowflake) {
 		: getMessage("SUCCESS_THREAD_CREATE", requestId);
 
 	if (msgContent && msgContent.length > 0) {
-		await thread.send({
+		const msg = await thread.send({
 			content: msgContent,
 			components: [buttonRow],
 		});
+
+		if (botMember.permissionsIn(thread.id).has(Permissions.FLAGS.MANAGE_MESSAGES)) {
+			await msg.pin();
+			await thread.lastMessage?.delete();
```

Comment:
can you promse.all this please

Reply:
No, the pinned message needs to be created before we try to delete the latest message.